core-rust/state-manager/src/store/address_book_components.rs

```diff
@@ -0,0 +1,40 @@
+use crate::engine_prelude::Sbor;
+
+
+/// The ID of the node stored in the Address book (Secp256k1 public key)
+#[derive(Clone, Copy, Sbor)]
+pub struct AddressBookNodeId(pub [u8; AddressBookNodeId::LENGTH]);
+
+impl AddressBookNodeId {
+    pub const LENGTH: usize = 33;
+
+    pub fn new(id: [u8; Self::LENGTH]) -> Self {
+        Self(id)
+    }
+
+    pub fn as_bytes(&self) -> &[u8; Self::LENGTH] {
+        &self.0
+    }
+}
+
+/// Peer address entry with all components
+#[derive(Clone, Sbor)]
+pub struct PeerAddress {
+    pub encoded_uri: Vec<u8>,
+    pub latest_connection_status: Option<ConnectionStatus>,
+    pub last_seen: Option<i64>,
+}
+
+#[derive(Clone, Copy, Sbor)]
+pub enum ConnectionStatus {
+    Connected,
+    Disconnected,
+}
+
+/// Address book entry
+#[derive(Clone, Sbor)]
+pub struct AddressBookEntry {
+    pub node_id: AddressBookNodeId,
+    pub banned_until: Option<i64>,
```

Comment:
If that's a timestamp we probably want to create an `EpochTimestamp` new-type wrapping e.g. chrono's [DateTime<Utc>](https://docs.rs/chrono/latest/chrono/struct.DateTime.html) - with this to make it SBOR encode as an i64:

```
#[sbor(
    as_type = "i64",
    as_ref = "self. timestamp()",
    from_value = "Self(DateTime::from_timestamp(value, 0))"
)]
```

Reply:
Thanks for the hint.

Reply:
After looking deeper, I realized that implementation must make assumptions about future use and convenient representation. Also, it requires adding a new dependency on `crono` just to make the code compile, there is no use of this representation on the Rust side for now. So, I've created a type alias and added a comment how this can be changed if necessary. Actual change, better to postpone until these timestamps will be actually used on the Rust side. Someone who will be implementing this part will be free to choose the most convenient underlying type without breaking binary compatibility.